### PR TITLE
Trento-change#7: Add new section "Rotating API keys"

### PR DESCRIPTION
**Pull request by tomschr**
### PR creator: Description

A new section about rotation API keys.

### PR creator: Are there any relevant issues/feature requests?

* change#7
* https://confluence.suse.com/x/_YA0WQ

### CODE REVIEW

**Comment by abravosuse** [comment]
@tomschr please review when you get a chance

File: trento/xml/article_sap_trento.xml [hunk 1]
```diff
@@ -954,7 +951,8 @@ As agreed on https://confluence.suse.com/x/DAEcN on our Trento doc kick off
         <formalpara>
           <title><guimenu>Settings</guimenu></title>
           <para>
-           Lets you retrieve the API key for this particular installation, which is required for the &t.agent; configuration.
+           Lets you retrieve and operate the API key for this particular installation, which is required for the &t.agent; configuration, as well as
+           managing the connection data to a SUSE Manager instance when one is available.
```
Comment (tomschr) [inline trento/xml/article_sap_trento.xml:955]:
```suggestion
           Lets you retrieve and operate the API key for this particular installation. This is required for the &t.agent; configuration, as well as
           managing the connection data for a &suma; instance when one is available.
```

File: trento/xml/article_sap_trento.xml [hunk 2]
```diff
@@ -2286,10 +2284,33 @@ In the Trento dashboard, go to the overview corresponding to the object for whic
...
+        details. </para> 
     </section>
   </section>
 -->
+
+  <section xml:id="sec-trento-rotating-apikeys">
+    <title>Rotating API keys</title>
+    <remark>toms 2024-04-18 (change#7)</remark>
+    <para>The communication from the Trento agent to the Trento server is secured by a API key,
+     which must be provided in the agent configuration file.</para>
```
Comment (tomschr) [inline trento/xml/article_sap_trento.xml:2296]:
```suggestion
    <para>The communication from the &t.agent; to the &t.server; is secured by a API key
     which must be provided in the agent configuration file.</para>
```

File: trento/xml/article_sap_trento.xml [hunk 3]
```diff
@@ -2286,10 +2284,33 @@ In the Trento dashboard, go to the overview corresponding to the object for whic
...
+   <para>By default, the API key does not have an expiration date. But, in order to increase the
+    overall security of the setup and/or meet internal security requirements,
+    the user can set up a custom expiration date.</para>
```
Comment (tomschr) [inline trento/xml/article_sap_trento.xml:2299]:
```suggestion
   <para>By default, the API key does not have an expiration date. To increase the
    overall security of the setup, meet internal security requirements, or both,
    the user can set up a custom expiration date.</para>
```

File: trento/xml/article_sap_trento.xml [hunk 4]
```diff
@@ -2286,10 +2284,33 @@ In the Trento dashboard, go to the overview corresponding to the object for whic
...
+   <para>In order to do that, the user should go to the Settings view and click on
+    the Generate Key button in the API Key section:</para>
```
Comment (tomschr) [inline trento/xml/article_sap_trento.xml:2301]:
```suggestion
   <para>To do that, go to the Settings view and click 
    the <guimenu>Generate Key</guimenu> button in the API Key section:</para>
```

File: trento/xml/article_sap_trento.xml [hunk 5]
```diff
@@ -2286,10 +2284,33 @@ In the Trento dashboard, go to the overview corresponding to the object for whic
...
+          <figure>
+        <title><guimenu>Checks catalog</guimenu></title>
+        <mediaobject>
+          <imageobject>
+            <imagedata width="80%"
+              fileref="trento-api-key-settings.png" />
+          </imageobject>
+        </mediaobject>
+      </figure>
+   <para>Beware that whenever a new key is generated, the configuration of all the reporting agents must be updated accordingly.</para>
```
Comment (tomschr) [inline trento/xml/article_sap_trento.xml:2311]:
```suggestion
   <para>Whenever a new key is generated, the configuration of all the reporting agents must be updated accordingly.</para>
```

**Review by tomschr — COMMENTED** [review]
Thanks @abravosuse! :+1: Just a few small comments.

**Pull request merged**